### PR TITLE
Merge demo apps

### LINKED ISSUE #540: Explore: Merge Views from SwiftUI demo app into UIKit demo app
Let's explore the idea of having one entry point to all the SwiftUI demo screens. So it will look like one cell from the main table view which will show the SwiftUI example app internally. So code wise should be everything inside SwiftUI context, and we avoid maintaining 2 different demo apps.

### PR DESCRIPTION
**Pull request by etoledom**
Closes #540

### CODE REVIEW

File: Demo/Demo/Gravatar-UIKit-Demo/MainTableViewController.swift [hunk 1]
```diff
@@ -64,6 +60,8 @@ class MainTableViewController: UITableViewController {
         case .imageCropper:
             content.text = "Image Cropper"
         #endif
+        case .swiftUI:
+            content.text = "Swift UI"
```
Comment (pinarol) [inline Demo/Demo/Gravatar-UIKit-Demo/MainTableViewController.swift]: nit: Seems like there should be no space here, like: SwiftUI

**Comment by pinarol** [comment]
Looking good so far. We can also rename "Gravatar UIKit" as "Gravatar Demo" at this point.

File: Demo/Demo/Gravatar-UIKit-Demo/MainTableViewController.swift (hunk 1 above)
Comment (etoledom) [reply Demo/Demo/Gravatar-UIKit-Demo/MainTableViewController.swift]: Hah true!

File: fastlane/Fastfile [hunk 2]
```diff
@@ -99,16 +98,14 @@ platform :ios do
   end
 
   desc 'Upload the demo app with the given name to App Center'
-  lane :upload_demo_to_appcenter do |name:, build_number:|
-    UI.user_error!("Unsupported name #{name}") unless %w[SwiftUI UIKit].include? name
-
-    slug = "Gravatar-SDK-Demo-#{name}"
-    upload_to_appcenter(appcenter_slug: slug, artifact_basename: "Gravatar #{name}")
+  lane :upload_demo_to_appcenter do |build_number:|
+    slug = "Gravatar-SDK-Demo"
```
Comment (dangermattic) [inline fastlane/Fastfile:102]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/Gravatar-SDK-iOS/blob/13f8530c8ba82c86c636a3db03f8379d20580d5b/fastlane/Fastfile#L102"></span>Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.</td>
    </tr>
  </tbody>
</table>

File: fastlane/Fastfile [hunk 3]
```diff
@@ -99,16 +98,14 @@ platform :ios do
...
+    upload_to_appcenter(appcenter_slug: slug, artifact_basename: "Gravatar Demo")
```
Comment (dangermattic) [inline fastlane/Fastfile:103]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/Gravatar-SDK-iOS/blob/13f8530c8ba82c86c636a3db03f8379d20580d5b/fastlane/Fastfile#L103"></span>Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.</td>
    </tr>
  </tbody>
</table>

File: fastlane/Fastfile [hunk 4]
```diff
@@ -99,16 +98,14 @@ platform :ios do
...
 
     next unless is_ci
 
-    annotate_pr_with_appcenter_link(app_suffix: name)
-    annotate_buildkite_with_appcenter_link(app_suffix: name, appcenter_slug: slug, build_number: build_number)
+    annotate_pr_with_appcenter_link()
```
Comment (dangermattic) [inline fastlane/Fastfile:107]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/Gravatar-SDK-iOS/blob/13f8530c8ba82c86c636a3db03f8379d20580d5b/fastlane/Fastfile#L107"></span>Style/MethodCallWithoutArgsParentheses: Do not use parentheses for method calls with no arguments.</td>
    </tr>
  </tbody>
</table>

File: fastlane/Fastfile [hunk 5]
```diff
@@ -179,22 +171,22 @@ def upload_to_appcenter(appcenter_slug:, artifact_basename:)
   )
 end
 
-def annotate_pr_with_appcenter_link(app_suffix:)
+def annotate_pr_with_appcenter_link()
   comment_body = prototype_build_details_comment(
-    app_display_name: "Gravatar #{app_suffix} Prototype Build",
+    app_display_name: "Gravatar Prototype Build",
```
Comment (dangermattic) [inline fastlane/Fastfile:176]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/Gravatar-SDK-iOS/blob/13f8530c8ba82c86c636a3db03f8379d20580d5b/fastlane/Fastfile#L176"></span>Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.</td>
    </tr>
  </tbody>
</table>

File: fastlane/Fastfile [hunk 6]
```diff
@@ -179,22 +171,22 @@ def upload_to_appcenter(appcenter_slug:, artifact_basename:)
...
     app_center_org_name: APPCENTER_OWNER_NAME,
     fold: true
   )
 
   comment_on_pr(
     project: GITHUB_REPO,
     pr_number: Integer(ENV.fetch('BUILDKITE_PULL_REQUEST', nil)),
-    reuse_identifier: "prototype-build-link-gravtar-#{app_suffix}",
+    reuse_identifier: "prototype-build-link-gravtar-demo",
```
Comment (dangermattic) [inline fastlane/Fastfile:184]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/Gravatar-SDK-iOS/blob/13f8530c8ba82c86c636a3db03f8379d20580d5b/fastlane/Fastfile#L184"></span>Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.</td>
    </tr>
  </tbody>
</table>

**Review by andrewdmontgomery — APPROVED** [review]
Looks great

**Review by pinarol — APPROVED** [review]

**Comment by pinarol** [comment]
Can we merge it right away? Or is there a reason to keep it waiting?

**Comment by pinarol** [comment]
One thing I am curious about, we can still have SwiftUI previews after doing this right?

**Comment by andrewdmontgomery** [comment]
We can merge this as is.  I was going to yesterday, but got caught up in something else.

This leaves two changes for later (from @etoledom):
- Bundle identifiers.
   - Since this will mess up with provisioning profiles and the app center config.
- The app center app name.
   - This probably would break other branches CIs

He was thinking (and I agreed) that, since those changes would be disruptive, next week might be a great time to make them.

**Comment by andrewdmontgomery** [comment]
> One thing I am curious about, we can still have SwiftUI previews after doing this right?

Yep

image

**Pull request merged**